pkg/util/util.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/docker/engine-api/types/container"
+)
+
+// SafeForLoggingContainerConfig returns a string version of the container.Config object
+// with sensitive information (proxy environment variables containing credentials)
+// redacted.
+func SafeForLoggingContainerConfig(config *container.Config) string {
+	origEnv := config.Env
+	strippedEnv := StripProxyCredentials(origEnv)
+	config.Env = strippedEnv
```

Comment:
operating on the original config object and not a shallow copy could in principle expose us to a race in future - suggest copying the config object up front